### PR TITLE
Add additional body class for the account dashboard (#30050)

### LINKED ISSUE #30050: New classes in body tag - for dashboard page and with custom endpoint name
**Prerequisites (mark completed items with an [x]):**
- [X] I have checked that my issue type is not listed here https://github.com/woocommerce/woocommerce/issues/new/choose
- [x] My issue is not a security issue, support request, bug report, enhancement or feature request (Please use the link above if it is).

**Issue Description:**

Hi,
when we browse my account pages all endpoints have specific classes in the body tag as in the examples below:

.woocommerce-account.woocommerce-orders

.woocommerce-account.woocommerce-downloads

but when we are on the my account panel page the class placed in the body tag is just .woocommerce-account
there is no other more specific, so to select elements on this page it ends up being by exclusion as in the example below:

.woocommerce-account:not(.woocommerce-orders):not(.woocommerce-downloads)

you could introduce the **.woocommerce-panel** or **.woocommerce-dashboard** class

### PR DESCRIPTION
**Pull request by medavidallsop**
### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

### Changes proposed in this Pull Request:

This adds a body class of `woocommerce-dashboard` to the account home page, for the reasons highlighted in #30050.

Closes #30050.

### How to test the changes in this Pull Request:

1. Use the Storefront theme
2. Ensure you are logged in to the frontend as a customer
3. Access the my account page (don't click any of the tabs)
4. Review the body classes on this page, you'll see an additional `woocommerce-dashboard` class (i've used this name because the default tab is labelled Dashboard)
5. Click into the other tabs of my account, on these pages you'll see that the body class is not included
6. Log out, on the login page for my account, you'll see that the body class is not included

### Changelog entry

-   [x] Automatically create a changelog entry from the details below.

-   [ ] This Pull Request does not require a changelog entry. (Comment required below)

<details>

<summary>Changelog Entry Details</summary>

#### Significance

-   [ ] Patch
-   [x] Minor
-   [ ] Major

#### Type

-   [ ] Fix - Fixes an existing bug
-   [ ] Add - Adds functionality
-   [ ] Update - Update existing functionality
-   [ ] Dev - Development related task
-   [ ] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [x] Enhancement - Improvement to existing functionality

#### Message 

Add additional body class for the account dashboard

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment 

</details>

### CODE REVIEW

File: plugins/woocommerce/includes/wc-template-functions.php [hunk 1]
```diff
@@ -327,6 +330,21 @@ function wc_body_class( $classes ) {
...
+		$account_page_id = get_option( 'woocommerce_myaccount_page_id' );
+		$account_page_slug = false;
+
+		if ( !empty( $account_page_id ) ) {
+
+			$account_page_slug = get_post_field( 'post_name', $account_page_id );
+
+		}
+
+		if ( basename( $wp->request ) == $account_page_slug && is_user_logged_in() ) {
```
Comment (Konamiman) [inline plugins/woocommerce/includes/wc-template-functions.php]: This `if` could be nested into the first one, right? Then the initial `$account_page_slug = false;` could be removed.

Comment (medavidallsop) [reply plugins/woocommerce/includes/wc-template-functions.php]: @Konamiman I have refactored this as per the feedback, please can you recheck?

**Comment by medavidallsop** [comment]
@Konamiman I noticed the small code refactor and the branch merge, can you let me know if you need anything further from me on this, thank you.

**Comment by Konamiman** [comment]
Hi @medavidallsop, nothing else needed in principle, it's just that it gets a bit tricky to get the branch up to date and to make all CI checks pass.

**Review by Konamiman — APPROVED** [review]

**Comment by welmartins** [comment]
Hello everyone \0/, thanks for implementing this request, it will help a lot in customizing this page, 😅 and I'm actually working on a project and I made a hack to add this class, by the way, you saw the other comments in the thread on #30050 about adding the class for custom endpoints:

Another thing, when creating custom endpoints, the endpoint name could be inserted as a class in the body tag too, as in the example below:

add_action( 'init', 'my_account_custom_endpoints' );
function my_account_custom_endpoints() {
add_rewrite_endpoint( 'my-endpoint', EP_ROOT | EP_PAGES );
}

< body class="woocommerce-account woocommerce-my-endpoint">

@medavidallsop @Konamiman

**Comment by medavidallsop** [comment]
@welmartins I think that adds a level of complexity that would need a little bit more investigation compared to the class we've just added, I would recommend you create a separate issue for that one with some sample code so whoever picks it up can replicate the custom setup and go from there, including some use case scenarios. Thank you.

**Pull request merged**